olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -790,6 +790,53 @@ bool VersionedLayerClientImpl::Release(const TileKeys& tiles) {
 
   return settings_.cache->Release(keys_to_release);
 }
+cache::KeyValueCache::KeyListType
+VersionedLayerClientImpl::GetRelatedPartitionKeys(
+    const std::string& partition_id) {
+  if (!settings_.cache) {
+    return {};
+  }
+
+  auto version = catalog_version_.load();
```

Comment:
This is a little wrong, as the version can be currently in the request phase so it will be soon available. Probably you should use `GetVersion` instead.

Reply:
Should I also change it for Protect/Release for tile keys, IsCached, RemoveFromCache, etc.?

Reply:
As these are all sync methods it is a little tricky. If we followed the same approach for all then we need to provide a good comment. So probably it is wiser to keep in sync with the rest and see if somebody will complain that this behaviour is indefinite. Instead please add a proper comment for this in the public API to this new APIs and to all related APIs. It is catalog version what is needed and not layer version. Also we need to specify by when it can be used, means that the version is set internally once one of the async APIs are called, e.g. `GetData`, `GetPartitions`, etc.

Reply:
For all sync methods keeping the same behavior, that version should be set in constructor, or offline request.